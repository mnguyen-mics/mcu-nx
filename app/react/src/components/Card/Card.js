import React, { Component, PropTypes } from 'react';
import { Row, Col, Button } from 'antd';

class Card extends Component {

  render() {

    const {
      title,
      buttons
    } = this.props;

    return (
      <Row className="mcs-table-container">
        <Row className="mcs-table-header mcs-card-header">
          <Col span={24}>
            <span className="mcs-card-title">{title}</span><span className="mcs-card-button">{buttons}</span>
          </Col>
        </Row>
        <Row className="mcs-table-body">
          <Col span={24}>
            {this.props.children}
          </Col>
        </Row>
      </Row>
    );
  }

}

Card.defaultProps = {
  buttons: <span />
};

Card.propTypes = {
  title: PropTypes.string.isRequired,
  buttons: PropTypes.element,
};

export default Card;
