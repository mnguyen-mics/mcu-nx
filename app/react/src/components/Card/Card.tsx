import * as React from 'react';
import { Row, Col, Spin } from 'antd';

export interface CardProps {
  buttons?: React.ReactNode;
  title?: React.ReactNode;
  isLoading?: boolean;
}

class Card extends React.Component<CardProps> {
  render() {
    const { title, buttons, isLoading, children } = this.props;

    const hasHeader = title || buttons;

    const titleElement = title && (
      <span className="mcs-card-title">{title}</span>
    );
    const buttonsElement = buttons && (
      <span className="mcs-card-button">{buttons}</span>
    );

    return (
      <Row className="mcs-card-container">
        {hasHeader && (
          <Row className="mcs-card-header">
            <Col span={24}>
              {titleElement}
              {buttonsElement}
            </Col>
            <Col span={24}>
              <hr />
            </Col>
          </Row>
        )}
        <Row>
          {isLoading ? (
            <Col span={24} className="text-center">
              <Spin />
            </Col>
          ) : (
            children
          )}
        </Row>
      </Row>
    );
  }
}

export default Card;
