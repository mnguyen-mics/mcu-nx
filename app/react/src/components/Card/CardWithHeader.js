import React from 'react';
import PropTypes from 'prop-types';
import { Row, Col } from 'antd';

import { HeaderItem } from './';

function CardWithHeader({ children, headerItems }) {

  const displayHeader = headerItems.map((element, i) => (
    <HeaderItem
      className={i !== headerItems.length - 1 ? 'section border' : 'section'}
      data={element}
      key={element.translationKey}
    />
  ));

  return (
    <Row className="mcs-table-container-full">
      <Row className="mcs-table-header">{ displayHeader }</Row>
      <Row className="mcs-table-body">
        <Col span={24}>
          {children}
        </Col>
      </Row>
    </Row>
  );
}

CardWithHeader.defaultProps = {
  buttons: <span />,
  hasHeader: true,
  headerItems: [],
};

CardWithHeader.propTypes = {
  headerItems: PropTypes.arrayOf(PropTypes.shape({
    iconType: PropTypes.string,
    translationKey: PropTypes.string,
    number: PropTypes.string,
  })),
};

export default CardWithHeader;
