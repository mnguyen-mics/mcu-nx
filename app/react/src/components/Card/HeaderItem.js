import React from 'react';
import PropTypes from 'prop-types';
import { Col } from 'antd';
import { FormattedMessage } from 'react-intl';

import McsIcons from '../McsIcons';

function HeaderItem({ className, data }) {

  return (
    <Col span={6} className={className}>
      <McsIcons type={data.iconType} />
      <div className="title"><FormattedMessage id={data.translationKey} /></div>
      <div className="number">{data.number}</div>
    </Col>
  );
}

HeaderItem.propTypes = {
  className: PropTypes.string.isRequired,

  data: PropTypes.shape({
    iconType: PropTypes.string,
    translationKey: PropTypes.string,
    number: PropTypes.string,
  }).isRequired,
};

export default HeaderItem;
