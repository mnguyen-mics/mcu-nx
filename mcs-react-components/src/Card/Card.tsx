import * as React from 'react';
import { Row, Col, Spin } from 'antd';

export interface CardProps {
  buttons?: JSX.Element;
  title?: string;
  isLoading?: boolean;
}

const Card: React.SFC<CardProps> = (props) => {
  const { title, buttons, isLoading, children } = props;
  return (
    <Row className="mcs-card-container">
      { (title || buttons)
        && <Row className="mcs-card-header">
          <Col span={24}>
            <span className="mcs-card-title">{title}</span>
            <span className="mcs-card-button">{buttons}</span>
          </Col>
          <Col span={24}><hr /></Col>
        </Row>
      }
      <Row>
        { isLoading ? (
          <Col span={24} className="text-center">
            <Spin />
          </Col>
        ) : children}
      </Row>
    </Row>
  );
};

Card.defaultProps = {
  buttons: undefined,
  title: undefined,
  isLoading: false,
};

export default Card;
