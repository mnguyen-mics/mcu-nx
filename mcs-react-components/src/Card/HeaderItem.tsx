import * as React from 'react';
import { Col } from 'antd';
import { FormattedMessage } from 'react-intl';
import McsIcons, { McsIconType } from '../McsIcons';

export interface HeaderItemDataProp {
  iconType: McsIconType;
  translationKey: string;
  number: number;
}
export interface HeaderItemProps {
  className?: string;
  data: HeaderItemDataProp;
}

const HeaderItem: React.SFC<HeaderItemProps> = props => {
  let className = '';
  if (props.className) className = props.className;
  return (
    <Col span={6} className={className}>
      <McsIcons type={props.data.iconType} />
      <div className="title" >
        <FormattedMessage id={props.data.translationKey} />
      </div>
      <div className="number">{props.data.number}</div>
    </Col>);
};

export default HeaderItem;
