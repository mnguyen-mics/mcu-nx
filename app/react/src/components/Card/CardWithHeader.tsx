import * as React from 'react';
import { Row, Col } from 'antd';
import HeaderItem, { HeaderItemDataProp } from './HeaderItem';

interface CardWithHeaderProps {
  headerItems?: HeaderItemDataProp[];
  buttons?: JSX.Element;
  hasHeader?: boolean;
}

const CardWithHeader: React.SFC<CardWithHeaderProps> = ({ children, headerItems }) => {

  const displayHeader = headerItems!.map((element, i) => (
    <HeaderItem
      className={i !== headerItems!.length - 1 ? 'section border' : 'section'}
      data={element}
      key={element.translationKey}
    />
  ));

  return (
    <Row className="mcs-table-container-full">
      <Row className="mcs-table-header">{displayHeader}</Row>
      <Row className="mcs-table-body">
        <Col span={24}>
          {children}
        </Col>
      </Row>
    </Row>
  );
};

CardWithHeader.defaultProps = {
  headerItems: [{
    iconType: 'warning',
    translationKey: '',
    number: 0,
  }],
  buttons: <span />,
  hasHeader: true,
};

export default CardWithHeader;
