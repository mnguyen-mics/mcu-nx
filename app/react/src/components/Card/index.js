import Card from './Card';
import CardWithHeader from './CardWithHeader';
import HeaderItem from './HeaderItem';

export {
  Card,
  CardWithHeader,
  HeaderItem,
};
