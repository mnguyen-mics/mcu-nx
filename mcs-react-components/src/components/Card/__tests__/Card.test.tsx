import 'jest';
import * as React from 'react';
import { Button } from 'antd';
import * as TestRenderer from 'react-test-renderer';
import Card, { CardProps } from '../Card';

it('renders the Card', () => {
  const props: CardProps = {
    buttons: <Button type="primary" href="">Test</Button>,
    title: 'Title',
  };
  const component = TestRenderer.create(
    <Card {...props}>
      <div>Content</div>
    </Card>,
  );
  const res = component.toJSON();
  expect(res).toMatchSnapshot();
});
