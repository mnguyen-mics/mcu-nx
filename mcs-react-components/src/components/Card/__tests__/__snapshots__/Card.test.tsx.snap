// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the Card 1`] = `
<div
  className="ant-row mcs-card-container "
  style={Object {}}
>
  <div
    className="ant-row mcs-card-header"
  >
    <div
      className="ant-col ant-col-24"
    >
      <span
        className="mcs-card-title"
      >
        Title
      </span>
      <span
        className="mcs-card-button"
      >
        <button
          className="ant-btn ant-btn-primary"
          onClick={[Function]}
          type="button"
        >
          <span>
            Test
          </span>
        </button>
      </span>
    </div>
  </div>
  <div
    className="ant-row"
    style={Object {}}
  >
    <div>
      Content
    </div>
  </div>
</div>
`;
